src/RA_StringUtils.h

```diff
@@ -496,24 +461,33 @@ class StringBuilder
 
     bool m_bPrepareWide;
 
-    struct PendingString
-    {
-        union {
-            const std::string* String;
-            const std::wstring* WString;
-
-            struct
-            {
-                const char* Pointer;
-                size_t Length;
-            } Char;
-
-            struct
-            {
-                const wchar_t* Pointer;
-                size_t Length;
-            } WChar;
-        } Ref{};
+    class PendingString
+    {
+    public:
+        explicit PendingString(std::string&& arg) noexcept : String{std::move(arg)}, DataType{Type::String} {}
+        explicit PendingString(std::wstring&& arg) noexcept : WString{std::move(arg)}, DataType{Type::WString} {}
+        explicit PendingString(_In_z_ const char* const restrict ptr, std::size_t len) noexcept :
+            Ref{std::string_view{ptr, len}},
+            DataType{Type::CharRef}
+        {}
+        explicit PendingString(_In_z_ const wchar_t* const restrict ptr, std::size_t len) noexcept :
+            Ref{std::wstring_view{ptr, len}},
+            DataType{Type::WCharRef}
+        {}
+        PendingString() noexcept = delete;
+        ~PendingString() noexcept = default;
+
+        // To prevent copying moving
+        PendingString(const PendingString&) = delete;
+        PendingString& operator=(const PendingString&) = delete;
+        PendingString& operator=(PendingString&&) noexcept = delete;
```

Comment:
No copying is ensured now. Move construction is still needed for perfect forwarding.